src/main/java/org/truffleruby/core/support/RandomizerNodes.java

```diff
@@ -284,27 +281,16 @@ private static long makeMask(long x) {
     @Primitive(name = "randomizer_gen_seed")
     public static abstract class RandomizerGenSeedPrimitiveNode extends PrimitiveArrayArgumentsNode {
 
-        // Single instance of Random per host VM
-        private static final Random RANDOM;
-
-        static {
-            try {
-                RANDOM = TruffleOptions.AOT ? SecureRandom.getInstance("SHA1PRNG") : new SecureRandom();
-            } catch (NoSuchAlgorithmException e) {
-                throw new UnsupportedOperationException(e);
-            }
-        }
-
-        @TruffleBoundary
         @Specialization
         public DynamicObject randomizerGenSeed(DynamicObject randomizerClass) {
-            final BigInteger seed = randomSeedBigInteger(RANDOM);
+            final BigInteger seed = randomSeedBigInteger(getContext().getRandom());
             return createBignum(seed);
         }
 
         private static final int DEFAULT_SEED_CNT = 4;
 
-        public static BigInteger randomSeedBigInteger(java.util.Random random) {
+        @TruffleBoundary
+        public static BigInteger randomSeedBigInteger(Random random) {
             byte[] seed = new byte[DEFAULT_SEED_CNT * 4];
             random.nextBytes(seed);
```

Comment:
Looks like this should use `SecureRandom#generateSeed()`, which could then be exposed by RubyContext so the SecureRandom can only be used for seeding.

Reply:
Since the number of bytes is always 16, we could also just store the seed directly in RubyContext.

Reply:
> Since the number of bytes is always 16, we could also just store the seed directly in RubyContext.

No, that would be incorrect, since this method can be called multiple times through Ruby Random.new_seed.

Reply:
We do not have enough entropy in practice to use `SecureRandom#generateSeed()` even in CI.

Reply:
That's unfortunate. Would be good to add a comment about it where `nextBytes()` is used as a workaround.

Reply:
I don't think it's used as you thought in the first place - it's used enough even during this like building C extensions that it obviously isn't intended to be from original entropy each time.

Reply:
FWIW it seems MRI uses `urandom` for `Random.new_seed`.